### PR TITLE
Add files to MessageEdit

### PR DESCRIPTION
**Pull request by ShadiestGoat**
This PR adds the ability to edit the attachments of a message, following the discord API.

### CODE REVIEW

File: message.go [hunk 1]
```diff
@@ -249,6 +249,10 @@ type MessageEdit struct {
 	Embeds          []*MessageEmbed         `json:"embeds"`
 	AllowedMentions *MessageAllowedMentions `json:"allowed_mentions,omitempty"`
 	Flags           MessageFlags            `json:"flags,omitempty"`
+	// Files to append to the message
+	Files []*File `json:"files,omitempty"`
+	// Overwrite existing attachments
+	Attachments *[]*MessageAttachment `json:"attachments,omitempty"`
```
Comment (ShadiestGoat) [inline message.go:255]: I made this a pointer because an empty array doesn't mean it should be not specified in the json.

**Comment by ShadiestGoat** [comment]
Forgot to mention: solves #1252

File: message.go [hunk 2]
```diff
@@ -249,6 +249,10 @@ type MessageEdit struct {
 	Embeds          []*MessageEmbed         `json:"embeds"`
 	AllowedMentions *MessageAllowedMentions `json:"allowed_mentions,omitempty"`
 	Flags           MessageFlags            `json:"flags,omitempty"`
+	// Files to append to the message
+	Files []*File `json:"files,omitempty"`
```
Comment (FedorLap2006) [inline message.go:253]: Files are sent using `MultipartBodyWithJSON`, therefore should be omitted from the request.
```suggestion
	Files []*File `json:"-"`
```

**Review by FedorLap2006 — CHANGES_REQUESTED** [review]

**Comment by ShadiestGoat** [comment]
@FedorLap2006 you're right, I changed it now

**Review by FedorLap2006 — APPROVED** [review]
LGTM. Though removing attachments won't work for now, since the `MultipartBodyWithJSON` encodes files a bit incorrectly.
I'll probably submit a fix for it soon.

**Comment by ShadiestGoat** [comment]
I don't want to be "that guy", but would it be possible to know a possible timeline for this to be merged?

**Comment by FedorLap2006** [comment]
Thanks for your contribution!

**Pull request merged**